src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -591,7 +614,7 @@ public Builder removeCustom(String type) {
         }
 
         public ClusterState build() {
-            return new ClusterState(clusterName, version, metaData, routingTable, nodes, blocks, customs.build());
+            return new ClusterState(clusterName, version, uuid, metaData, routingTable, nodes, blocks, customs.build());
```

Comment:
Random thought I want to make sure is not lost :) - does it make sense to assert that uuid is not `_na_` by the time we get here?